gramps_webapi/api/resources/emit.py

```diff
@@ -0,0 +1,94 @@
+"""Gramps Json Encoder."""
+
+import inspect
+
+import gramps.gen.lib as lib
+from flask import Response
+from flask.json import JSONEncoder
+from gramps.gen.db import DbBookmarks
+
+PRIMARY_CLASS_MAP = {
+    "Person": lib.Person,
+    "Family": lib.Family,
+    "Event": lib.Event,
+    "Place": lib.Place,
+    "Source": lib.Source,
+    "Citation": lib.Citation,
+    "Repository": lib.Repository,
+    "Media": lib.Media,
+    "Note": lib.Note,
+    "Tag": lib.Tag,
+}
+
+
+class GrampsJSONEncoder(JSONEncoder):
+    """Customizes Gramps Web API output."""
+
+    def __init__(self):
+        """Initialize class."""
+        JSONEncoder.__init__(self)
+        self.sort_keys = True
+        self.ensure_ascii = False
+        self.strip_empty_keys = False
+        self.filter_only_keys = []
+        self.filter_skip_keys = []
+        self.gramps_classes = [
+            getattr(lib, key) for key, value in inspect.getmembers(lib, inspect.isclass)
+        ]
+
+    def response(self, payload):
+        """Prepare response."""
+        return Response(
+            response=self.encode(payload),
+            status=200,
+            mimetype="application/json",
+        )
+
+    def api_filter(self, obj):
+        """Filter data for a Gramps object."""
+        data = {}
+        filter = False
+        try:
+            if isinstance(obj, PRIMARY_CLASS_MAP[self.gramps_class_name]):
+                filter = True
+        except:
+            pass
+        for key, value in obj.__dict__.items():
+            if filter:
+                if self.filter_only_keys != [] and key not in self.filter_only_keys:
+                    continue
+                if self.filter_skip_keys != [] and key in self.filter_skip_keys:
+                    continue
+            if key.startswith("_"):
+                key = key[2 + key.find("__") :]
+            if self.strip_empty_keys:
+                if isinstance(value, lib.GrampsType):
+                    data[key] = str(value)
+                elif isinstance(value, lib.StyledText):
+                    data[key] = str(value)
+                elif isinstance(value, lib.PlaceName):
+                    data[key] = {
+                        "date": value.date,
+                        "lang": value.lang,
+                        "value": value.value,
+                    }
```

Comment:
I can do that I guess.

Reply:
I'm not really sure what that accomplishes.

Reply:
Yeah maybe I'm being a bit extreme here. But for instance there might be several places in the code where there is a key indicating a locale string or language, and to avoid having "lang" in one place, "locale" in another, and maybe "language" somewhere else, using a constant for everything makes it easier to remain consistent. It also more easily catches typos (since a typo in a variable raises an error).

Reply:
Yeah this wasn't needed anymore in latest pass. Did make the character substitution for sex aka gender constants as suggested.